lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_download_translations_action.rb

```diff
@@ -0,0 +1,110 @@
+# This action is the new version of android_update_metadata (AndroidUpdateMetadataAction) and should now be used instead of that one
+
+module Fastlane
+  module Actions
+    class AndroidDownloadTranslationsAction < Action
+      def self.run(params)
+        require_relative '../../helper/android/android_localize_helper.rb'
+        require_relative '../../helper/git_helper.rb'
+
+        res_dir = File.join(ENV['PROJECT_ROOT_FOLDER'] || '.', params[:res_dir])
+
+        Fastlane::Helper::Android::LocalizeHelper.create_available_languages_file(
+          res_dir: res_dir,
+          locale_codes: [params[:source_locale]] + params[:locales].map { |h| h[:android] }
```

Comment:
Highlighting this will crash in `create_available_languages_file` if the locale is missing the android tag, do you think it's worth either printing a message or skipping the locale in this case vs crashing the lane?

Reply:
Crash will happen here 

https://github.com/wordpress-mobile/release-toolkit/pull/229/files#diff-06b039b5ae6274c91666cdec42e2d9895a71fcb64572578277ade0602e8da15fR190

Reply:
@JavonDavis Fixed via 84d6883

With that change (and new `verify_block` to check the parameter), here is what happens if I try to pass a locale map for which one of the entry doesn't have an `:android` key: 

```
[19:22:25]: Driving the lane 'android download_translations' 🚀
[19:22:25]: Error setting value '[{:glotpress=>"ar", :android=>"ar", :google_play=>"ar", :promo_config=>{}}, {:glotpress=>"de", :android=>"de", :google_play=>"de-DE", :promo_config=>{}}, {:glotpress=>"es", :android=>"es", :google_play=>"es-ES", :promo_config=>{}}, {:glotpress=>"fr", :android=>"fr", :google_play=>"fr-FR", :promo_config=>{}}, {:glotpress=>"he", :android=>"he", :google_play=>"iw-IL", :promo_config=>{}}, {:glotpress=>"id", :android=>"id", :google_play=>"id", :promo_config=>{}}, {:glotpress=>"it", :androisd=>"it", :google_play=>"it-IT", :promo_config=>{}}, {:glotpress=>"ja", :android=>"ja", :google_play=>"ja-JP", :promo_config=>{}}, {:glotpress=>"ko", :android=>"ko", :google_play=>"ko-KR", :promo_config=>{}}, {:glotpress=>"nl", :android=>"nl", :google_play=>"nl-NL", :promo_config=>{}}, {:glotpress=>"pt-br", :android=>"pt-rBR", :google_play=>"pt-BR", :promo_config=>{}}, {:glotpress=>"ru", :android=>"ru", :google_play=>"ru-RU", :promo_config=>{}}, {:glotpress=>"sv", :android=>"sv", :google_play=>"sv-SE", :promo_config=>{}}, {:glotpress=>"tr", :android=>"tr", :google_play=>"tr-TR", :promo_config=>{}}, {:glotpress=>"zh-cn", :android=>"zh-rCN", :google_play=>"zh-CN", :promo_config=>{}}, {:glotpress=>"zh-tw", :android=>"zh-rTW", :google_play=>"zh-TW", :promo_config=>{}}]' for option 'locales'
[19:22:25]: You passed invalid parameters to 'android_download_translations'.
[19:22:25]: Check out the error below and available options by running `fastlane action android_download_translations`
+------------------+-------------------------------+
|                   Lane Context                   |
+------------------+-------------------------------+
| DEFAULT_PLATFORM | android                       |
| PLATFORM_NAME    | android                       |
| LANE_NAME        | android download_translations |
+------------------+-------------------------------+
[19:22:25]: The value for the `locales` parameter must be an Array of Hashes, and each Hash must have at least `:glotpress` and `:android` keys.

+------+------------------+-------------+
|           fastlane summary            |
+------+------------------+-------------+
| Step | Action           | Time (in s) |
+------+------------------+-------------+
| 1    | default_platform | 0           |
+------+------------------+-------------+

[19:22:25]: fastlane finished with errors

[!] The value for the `locales` parameter must be an Array of Hashes, and each Hash must have at least `:glotpress` and `:android` keys.
```

FYI, to test this, I purposedly misspelled `:androisd: "it"` in [the draft branch I use for testing this PR on woocommerce-android](https://github.com/woocommerce/woocommerce-android/pull/3821) then ran the lane with that typo. You can see the subtle typo in the param in the log above (it is logged on the "Error setting value: …" line).

Reply:
Perfect! Thanks @AliSoftware 🚢